File: linear_search/linear_search.rb
# iterative implementation of linear search in Ruby

def linear_search(item, an_array)
    i = 0
    while i < an_array.length
        if item == an_array[i]
            return "#{item} found at position #{i}"
        end
        i += 1
    end
    return "#{item} not found in the list"
end

File: linear_search/linearSearch.js
// recursive linear search implementation
const search = (v, l, item) => {
  const n = v.length;
  if (l > n) {
    return false;
  }
  if (v[l] === item) {
    return true;
  }
  return search(v, l + 1, item);
}

const linearSearch = (v, item) => {
  return search(v, 0, item);
}

console.log(linearSearch([1, 5, 10, 79, 30, 6], 6));
